### DISCUSSION TITLE
Software version vs Platforms

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#8117] [Q&A]
I was looking to document the software installed on our servers into netbox.
Seems that a Platform should be the way to go for this.

but I have a question on this

* When I set the Manufacturer, it also limits the platform to only be visibl to devices with the same Manufacturer.
This looks a strange limitation to me. When I create a 'Manufacturer': Oracle and a Name 'Linux R8'; then I would like to be able to assing that to a device with Manufacturer 'Dell' or 'HP'

* Missing a 'version' field
A separate field for the 'version' or 'build' of the OS would be helpfull. It makes it more easy to filter and search

Pieter

**Comment by candlerb** [comment]
> When I create a 'Manufacturer': Oracle and a Name 'Linux R8'; then I would like to be able to assing that to a device with Manufacturer 'Dell' or 'HP'

For things like Linux distributions, you need to leave the manufacturer set to Null on the platform; then the platform is available on *any* device type from any manufacturer.

There is no option to link a platform to a subset of manufacturers, but even if you could, that wouldn't always solve the problem.  Dell make both servers and switches.  Cisco have both IOS and NXOS, but you can't use both on the same device type.  (A more sophisticated solution might be to have a "Platform Group", and each individual Device Type and Platform links to a Platform Group. Then "x86_64" would be one such platform group)

As for versions, it's currently suggested that you enumerate the major versions as separate platforms (e.g. "Ubuntu 18.04", "Ubuntu 20.04").  If you had sub-versions for each version, then it would be essentially the same but flattened out.

Speaking for myself, I don't find the need to model sub-versions as different platforms, say Ubuntu 20.04.1 and 20.04.2, or RHEL 8.0 and 8.1. These point releases are generally just in-place upgrades with accumulated bugfixes, and don't affect the types of applications of configs that can be pushed out to them.  So "Ubuntu 20.04" or "RHEL 8" is sufficient for my needs.

If you find you need a *text* field for the sub-version, then I suggest you try adding it as a custom field, linked to DCIM>Device and Virtualization>VirtualMachine.  See how it works out for you, and if you think it's generally useful, you can raise it as a FR.  You may find that a custom field is sufficient as the final solution, except that searching and filtering on custom fields (including via API) is limited at the moment.

**Reply by PieterL75** [reply]
Brian,

I was thinking of using it the same way as you propose.
but then I was wondering why the Manufacturer has to be the same then the device, hence my question :-)

It still makes sense to me to set 'Oracle' or 'Microsoft' in the Manufacturer field and 'Linux R8' or ' Windows Server 2027' to the Name field.

Or maybe I'm just too conservative when trying to migrate from our current CMDB to Netbox.

**Comment by candlerb** [comment]
> but then I was wondering why the Manufacturer has to be the same then the device, hence my question :-)

It's so that certain platforms which are only valid for (say) "Cisco" devices, are only shown for devices manufactured by "Cisco".  It's a way of pruning the list of choices offered when you choose the platform for a device.

However it doesn't really work for x86_64 servers, where there can be many manufacturers which can run the same software.  So for those platforms, you should leave the manufacturer unset.

It's *not* intended to reflect the manufacturer of the platform (e.g. "Microsoft" for "Windows") - it's the manufacturer of the devices that this platform can be selected on.
